common-lib/src/main/scala/com/gu/mediaservice/lib/aws/S3.scala

```diff
@@ -29,10 +31,39 @@ class S3(credentials: AWSCredentials) {
   lazy val client: AmazonS3 =
     new AmazonS3Client(credentials) <| (_ setEndpoint s3Endpoint)
 
-  def signUrl(bucket: Bucket, url: URI, expiration: DateTime): String = {
+  private def removeExtension(filename: String): String = {
+    val regex = """\.[a-zA-Z]{3,4}$""".r
+    regex.replaceAllIn(filename, "")
+  }
+
+  private def getContentDispositionFilename(image: Image, charset: CharSet): String = {
+    val baseFilename: String = image.uploadInfo.filename match {
+      case Some(f)  => s"${removeExtension(f)} (${image.id}).jpg"
+      case _        => s"${image.id}.jpg"
+    }
+
+    charset match {
+      case CharSet.UTF8 => {
+        // URLEncoder converts ` ` to `+`, replace it with `%20`
+        // See http://docs.oracle.com/javase/6/docs/api/java/net/URLEncoder.html
+        URLEncoder.encode(baseFilename, CharSet.UTF8).replace("+", "%20")
+      }
+      case characterSet => baseFilename.getBytes(characterSet).toString
+    }
+  }
+
+  def signUrl(bucket: Bucket, url: URI, image: Image, expiration: DateTime): String = {
     // get path and remove leading `/`
     val key: Key = url.getPath.drop(1)
-    val request = new GeneratePresignedUrlRequest(bucket, key).withExpiration(expiration.toDate)
+
+    // use both `filename` and `filename*` parameters for compatibility with user agents not implementing RFC 5987
+    // they'll fallback to `filename`, which will be a UTF-8 string decoded as Latin-1
```

Comment:
Can we leave a note saying that yes it will be garbage, just garbage that the browsers will accept?